frontend/src/metabase/components/form/CustomForm/CustomFormSubmit.tsx

```diff
@@ -22,12 +23,20 @@ function CustomFormSubmit({
   renderSubmit,
   disablePristineSubmit,
   children,
+  shouldPersistError,
+  fields,
+  error,
   ...props
 }: CustomFormSubmitProps & FormLegacyContext) {
   const title = children || submitTitle || t`Submit`;
   const canSubmit = !(
     submitting ||
-    invalid ||
+    isFormInvalid({
```

Comment:
This way, I only change the form that set `shouldPersistError: true`'s behavior. Normally, after any field is visited the error messages would be cleared.

#### Note

The loom video is a bit outdated, now the submit button should be enabled right after any field is modified, not when the field is blurred after being modified.

https://www.loom.com/share/8728395752504fc58adadac7c1e21130 

p.s. the logic is a bit confusing as everything combined is negated at the end. Actually, it should have read *Disable submit button when all fields are untouched (after form submission)*